nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ListDatabaseTables.java

```diff
@@ -303,8 +303,8 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                     stateMapProperties.put(fqn, Long.toString(System.currentTimeMillis()));
                 }
             }
-            // Update the last time the processor finished successfully
-            stateManager.replace(stateMap, stateMapProperties, Scope.CLUSTER);
+            // Update the timestamps for listed tables
+            stateManager.setState(stateMapProperties, Scope.CLUSTER);
```

Comment:
I would not change this to setState. It will overwrite anything that is in State. If someone does end up running clustered and not primary Node it will blow it away without warnings. 

Instead just check if the prior map version was -1 and do a set instead of replace.